src/templates/TemplateData.ts

```diff
@@ -111,146 +122,114 @@ export class TemplateData {
     }
 }
 
-function verifyTemplatesByRuntime(templatesMap: { [runtime: string]: Template[] }, runtime: ProjectRuntime): void {
+function verifyTemplatesByRuntime(templates: Template[], runtime: ProjectRuntime): void {
     if (runtime === ProjectRuntime.one) {
         for (const verifiedTemplateId of verifiedTemplates) {
-            if (!templatesMap[ScriptProjectCreatorBase.defaultRuntime].some((t: Template) => t.id === verifiedTemplateId)) {
+            if (!templates.some((t: Template) => t.id === verifiedTemplateId)) {
                 throw new Error(localize('failedToFindJavaScriptTemplate', 'Failed to find verified template with id "{0}".', verifiedTemplateId));
             }
         }
 
         for (const verifiedTemplateId of verifiedCSharpTemplates) {
-            if (!templatesMap[ProjectRuntime.one].some((t: Template) => t.id === verifiedTemplateId)) {
+            if (!templates.some((t: Template) => t.id === verifiedTemplateId)) {
                 throw new Error(localize('failedToFindCSharpTemplate', 'Failed to find verified template with id "{0}".', verifiedTemplateId));
             }
         }
     } else if (runtime === ProjectRuntime.beta) {
         for (const verifiedTemplateId of verifiedCSharpTemplates) {
-            if (!templatesMap[ProjectRuntime.beta].some((t: Template) => t.id === verifiedTemplateId)) {
+            if (!templates.some((t: Template) => t.id === verifiedTemplateId)) {
                 throw new Error(localize('failedToFindCSharpTemplate', 'Failed to find verified template with id "{0}".', verifiedTemplateId));
             }
         }
     }
 }
 
-export async function tryGetTemplateDataFromCache(reporter: TelemetryReporter | undefined, globalState: vscode.Memento, cliFeedJson: cliFeedJsonResponse): Promise<TemplateData | undefined> {
-    try {
-        return <TemplateData | undefined>await callWithTelemetryAndErrorHandling('azureFunctions.tryGetTemplateDataFromCache', reporter, undefined, async function (this: IActionContext): Promise<TemplateData | undefined> {
+export async function getTemplateData(globalState?: vscode.Memento): Promise<TemplateData> {
+    const templatesMap: { [runtime: string]: Template[] | undefined } = {};
+    const configMap: { [runtime: string]: Config | undefined } = {};
+    const cliFeedJson: cliFeedJsonResponse | undefined = await tryGetCliFeedJson();
+    return <TemplateData>await callWithTelemetryAndErrorHandling('azureFunctions.tryGetParsedTemplateDataFromCliFeed', ext.reporter, undefined, async function (this: IActionContext): Promise<TemplateData> {
```

Comment:
Let's move callWithTelemetryAndErrorHandling inside the for-loop so that we get  templateSource/errors per runtime